### PR TITLE
Formatting numbers

### PR DESCRIPTION
**Pull request by AuroraLantean**
fix input amount checking with count_decimals()
set output balances to 8dp, format prices according to Binance decimal rule

### CODE REVIEW

**Comment by AuroraLantean** [comment]
The design pattern is here: https://github.com/orgs/ggxchain/projects/6/views/1?pane=issue&itemId=60060225

File: src/components/dex/tokenSelector.tsx [hunk 1]
```diff
@@ -70,17 +73,24 @@ export default function TokenSelector({
...
 
 	const handleAmountChange = (e: ChangeEvent<HTMLInputElement>) => {
-		if (Number(e.target.value) > 10000000) {
+		let input = e.target.value;
+		const dpLen = count_decimals(input)
+		if(dpLen > MAX_DP) {
+			input = fixDP(input)
+		}
+		const num = Number(input);
+		if(Number.isNaN(num) || num < 0 || num > 10000000) {
```
Comment (Warchant) [inline src/components/dex/tokenSelector.tsx:86]: this check "isNan || num < 0 || num > 100000000" appears often, maybe make a helper func?

File: src/app/bridge-btc/page.tsx [hunk 2]
```diff
@@ -194,15 +194,20 @@ const BridgeBtc = () => {
...
-    const out = Number.parseInt(value);
-    //TODO handle parseFloat instead
-    if (Number.isNaN(out)) {
-      console.error('parseInt failed');
-    } else {
-      setAmountIp(out);
+    let input = event?.target.value;
+    lg('handleAmountChange:', input);
+
+    const dpLen = count_decimals(input)
```
Comment (akorchyn) [inline src/app/bridge-btc/page.tsx:200]: can be extracted as amountFormatter(string): number | undefined

**Review by akorchyn — APPROVED** [review]

File: src/app/bridge-btc/page.tsx (hunk 2 above)
Comment (AuroraLantean) [reply src/app/bridge-btc/page.tsx:200]: `input` line needs to be there to be passed down further.
`console.log` or `lg` is for current debugging until production version.
`const dpLen` needs to be there to compare with MAX_DP then modify the `input` right there.
I have made a `checkNumInput()` to extract some code. okay?

File: src/components/dex/tokenSelector.tsx (hunk 1 above)
Comment (AuroraLantean) [reply src/components/dex/tokenSelector.tsx]: I have made a checkNumInput() to those code.

**Pull request merged**